### ISSUE TITLE
Custom babel settings do not work

### ISSUE THREAD

**Issue by m-brantheim** [harrysolovay/rescripts#79]
I am making an app with Create-React-App, electron, rescripts. The setup has previously been working, but now I need a custom babel config to add decorators and I can't get it to work.

.rescriptsrc.js:
`module.exports = [
  require.resolve("./webpack.config.js"),
  ["use-babel-config", ".babelrc"]
]`

Was previously only the webpack.

webpack.config.js:
`module.exports = config => {
  config.target = "electron-renderer"
  return config
}`

.babelrc:
`{
  "presets": ["react-app"],
  "plugins": [
    ["@babel/plugin-proposal-decorators", { "legacy": true }],
    ["@babel/plugin-proposal-class-properties", { "loose": true }]
  ]
}`
I want to keep the CRA preset but add decorator support.

package.json:
`"devDependencies": {
    "@babel/core": "^7.6.4",
    "@babel/plugin-proposal-class-properties": "^7.7.4",
    "@babel/plugin-proposal-decorators": "^7.7.4",
    "@rescripts/cli": "^0.0.11",
    "@rescripts/rescript-env": "^0.0.10",
    "@rescripts/rescript-use-babel-config": "^0.0.9",
    "@storybook/addon-actions": "^5.2.5",
    "@storybook/addon-links": "^5.2.5",
    "@storybook/addons": "^5.2.5",
    "@storybook/react": "^5.2.5",
    "babel-loader": "^8.0.6",
    "babel-preset-react-app": "^9.1.0",
    "concurrently": "^4.1.2",
    "cross-env": "^5.2.1",
    "electron": "^6.0.3",
    "electron-builder": "^21.2.0",
    "eslint-config-airbnb": "^18.0.1",
    "eslint-config-prettier": "^6.5.0",
    "eslint-plugin-jsx-a11y": "^6.2.3",
    "eslint-plugin-prettier": "^3.1.1",
    "prettier": "^1.19.1",
    "prettier-eslint": "^9.0.1",
    "prettier-eslint-cli": "^5.0.0",
    "prettier-stylelint": "^0.4.2",
    "stylelint": "^12.0.0",
    "stylelint-config-standard": "^19.0.0",
    "wait-on": "^3.3.0"
  }
`
Error output in electron:
Element type is invalid: expected a string (for built-in components) or a class/function (for composite components) but got: undefined. You likely forgot to export your component from the file it's defined in, or you might have mixed up default and named imports.

Check the render method of `Icon`.  

If I remove the ["use-babel-config", ".babelrc"], then it works as normal but I cannot use decorators.
I think it is reading the .babelrc config because if I misstype something there I can crash the app. I have tried removing the "plugin-proposal-class-properties" but it does not make a difference.

Any ideas on something I might have missed?

**Comment by harrysolovay** [comment]
In using it, try wrapping your custom rescript with braces:

`webpack.config.js`

```js
module.exports = [
  ["use-babel-config", ".babelrc"] ,
  [require.resolve("./webpack.config.js")]
];
```

Also, above you wrote your webpack rescript as `module.exports = config => { config.target = "electron-renderer" return config }`. I want to make sure you're appropriately distinguishing between the two expressions in the function body. Can you try this:

`webpack.config.js`

```js
module.exports = config => {
  config.target = "electron-renderer";
  return config;
};
```

If all else fails, try reversing the order:

`.rescriptsrc.js`

```js
module.exports = [
  ["use-babel-config", ".babelrc"] ,
  require.resolve("./webpack.config.js")
];
```

If it's still broken, please share the code & I'll debug :)

**Comment by harrysolovay** [comment]
@m-brantheim closing this out since I haven't heard back from you.

**Comment by m-brantheim** [comment]
Hello again,

Sorry I was away for the holiday, just got back to work.

The braces was not the problem, I created a project from scratch using this config and it worked so I figured it must come from something in my code. I was able to track it down.

I have a function for importing .svg files to render them with react...something like this.
```javascript
import React from "react"
import { ReactComponent as AddIcon } from "./svg/Add.svg"
...
import IconEnum from "./IconEnum"

const getIcon = code => {
  switch (code) {
    case IconEnum.ADD:
      return <AddIcon />
    ...
  }
}

export default getIcon

```

Any idea why this breaks when I add ["use-babel-config", ".babelrc"]?
